### ISSUE TITLE
Introduce SciView Object?

### ISSUE THREAD

**Issue by kephale** [scenerygraphics/sciview#77]
Would it be worth it to make a SciView Object/Element/Node type? It would inherit from a Scenery Node, but be able to generate its own IJ2 dialogs for creation/adding, editing, and deleting based on its state and the state of the scenegraph. It could also be used for default implementations of SciJava Event handling, and other SciJava and ImageJ integration.

**Comment by skalarproduktraum** [comment]
Why do you need an own element type for that?
One thing I had in mind was that Nodes could get tags, would that suit your purpose?

**Comment by kephale** [comment]
Yay, discussion. 

I ran into this idea in a few different ways:
1. Setting up some physics simulations, which required a bunch of metadata for the physics engine
2. There is the SciJava ObjectService which could be leveraged for managing things that are both inside and outside of scenery (I had a better idea of examples for this at one point, but they will come back to me).

There is the metadata field in Node which could resolve (1), and I'm open to that. However, when I started reviving more of my old physics engine code I ran into some unpleasantness when it came to synchronizing geometries (IIRC that is when I posted the issue about getting access to vertex data from HasGeometry). Now that we have imagej-mesh I'm much more keen on using imagej-mesh structures for managing geometries. 

Let's keep this discussion going, tbh this was a pretty high priority for me before travel/grants/etc. kicked in.  Since then I haven't looked at the code that motivated this idea. I think this direction might actually lead to a SciJava object that contains a Node.

From my perspective there are a bunch of general-purpose really broadly applicable things that you get essentially for free when deriving from SciJava land, if things are done properly. Many are things that can be replicated by re-coding things, but the SciJava way is to maximize reuse. 

Just to be completely clear, the spirit of this idea isn't to introduce another type. At the moment Scenery exists in, essentially, a different universe from SciJava (aside from using the SciJava interpreter), and what I would like to see is a container that unifies Scenery Nodes and SciJava Objects/Types within the scope of SciView so we have less conversion code in SciView and get more benefits from the SciJava ecosystem. 

I suspect the above might induce some frustration on your side, and my preemptive response is that I am completely open to being convinced of favoring Scenery as a more central hub for objects/data within SciView when I read the chapter of your thesis that presents this argument (and I will delay on pursuing this implementation until then or it becomes absolutely essential, at which point I probably would have poked you multiple times anyway).
